### PR TITLE
[enhancement] Store windows basekit with symlink-preserving compression for nightly-build

### PR DESCRIPTION
**Pull request by icfaust**
## Description

actions/upload-artifact#625 included in 4.4.1 fixes an issue to not preserve symlinks (since fixed in later versions). The Windows Intel oneAPI BaseKit installer uses absolute paths in symlinks rather than relative ones for the ```latest``` folders of various oneAPI components.  This makes a maintenance burden on sklearnex-side since they are unusable. Therefore the fully path must be known and maintained to the compiler and tbb version```vars.bat``` files. A rather long and complicated batch script line was added to force these to be relative.  This removed the maintenance burden by fixing the ```latest``` symlink to work in sklearnex out-of-the-box.  Since it is only one line and fixes what hopefully will only be a temporary issue  I don't think necessitates its own script.  Because of the complication of it, comments were added for future understanding.

This will reduce maintenance burden on sklearnex side by making the CI pipeline run independent of the tbb/mkl/compiler versions

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Comment by icfaust** [comment]
Example run using matching main from my fork: https://github.com/icfaust/oneDAL/actions/runs/11263646195

**Review by Alexandr-Solovev — APPROVED** [review]
LGTM, just one question why the upload-artifact version has been decreased?

**Review by homksei — APPROVED** [review]

**Comment by icfaust** [comment]
Latest run showing valid operation of the nightly-build: https://github.com/icfaust/oneDAL/actions/runs/11373066491

**Comment by icfaust** [comment]
Here is the latest run on my scikit-learn-intelex fork with the successful run https://github.com/icfaust/scikit-learn-intelex/actions/runs/11378786935

**Comment by icfaust** [comment]
> LGTM, just one question why the upload-artifact version has been decreased?

Good question, the version without the later specifiers will now use the latest version of v4, I thought this would be nicer to prevent a large number of renovate-bot PRs for upgrading it.  They have corrected the symlink issues with the allowance of relative symlinks allowing for the builds to be as they were, so the version pin to 4.4.0 can be removed.
